File: project-6-bughunt/js/repro/repro.mjs
// Repro script: one check per injected bug. Exit 1 if any FAIL.
import { page, getProduct, promoExpires, total, checkout, CATALOG } from "../src/cart.mjs";

const results = [];
function check(name, cond) {
  results.push([name, cond]);
  console.log(cond ? "PASS" : "FAIL", name);
}

check("J1 page has full size", page([1, 2, 3, 4, 5, 6], 1, 2).length === 2);
const p = getProduct("widget");
p.price = 999;
check("J2 catalog shielded", CATALOG.widget.price === 10.0);
check("J3 promo expires Dec 2026", promoExpires().getFullYear() === 2026 && promoExpires().getMonth() === 11);
check("J4 total is a number", (await total(100)) === 107);
check("J5 checkout adds fee", checkout("widget", 2) === 25);

const failed = results.filter(([, ok]) => !ok).length;
console.log(`${results.length - failed}/${results.length} passed`);
process.exit(failed ? 1 : 0);

File: project-6-bughunt/python/repro/repro.py
"""Repro script: one check per injected bug. Exit 1 if any FAIL."""
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "src"))

os.environ.pop("APP_DEBUG", None)  # hunter runs with a clean env

from orders import add_note, bulk_rate, line_total, promo_active, receipt

results = []


def check(name, cond):
    results.append((name, cond))
    print(("PASS" if cond else "FAIL"), name)


check("B1 bulk_rate(100) == 0.20", bulk_rate(100) == 0.20)
check("B2 add_note isolated", add_note("a") == ["a"] and add_note("b") == ["b"])
check("B3 line_total(0.1, 3) == 0.3", line_total(0.1, 3) == 0.3)
check("B4 receipt hides cost by default", "cost" not in receipt("widget", 1))
check("B5 promo over year boundary", promo_active("01/05/2026") and not promo_active("01/05/2027"))

failed = [n for n, ok in results if not ok]
print(f"{len(results) - len(failed)}/{len(results)} passed")
sys.exit(1 if failed else 0)
